### PR TITLE
feat: Add test axis value column to summary table

### LINKED ISSUE #983: Add new column to matrix output table
**Author the user story for this feature**

As a User, I want to show current configuration when printing matrix table so I can see all informations about my finished tests.

**Is your feature request related to a problem? Please describe.**
In Gcloud matrix table contains `outcome`-`test_axis_value`-`outcome details`.
image

In Flank we could run multiple matrices at the same time, so our table contains `outcome`-`matrix id`-`outcome details`.
```
┌─────────┬──────────────────────┬───────────────────────────────┐
│ OUTCOME │      MATRIX ID       │         TEST DETAILS          │
├─────────┼──────────────────────┼───────────────────────────────┤
│  flaky  │ matrix-1aspi39ikq5hy │ 32 test cases passed, 8 flaky │
└─────────┴──────────────────────┴───────────────────────────────┘
```
However multiple configurations could be used to run on a single matrix and it should be shown in the table

**Describe the solution you'd like**
Add new column `test axis value` or `configuration` to show current configuration, so our table will have `outcome`-`matrix id`-`text axis value`-`outcome details`

With this approach, we will merge cloud details printing with Flank extended matrix report showing.

If you need more info contact with @jan-gogo  or @piotradamczyk5

### PR DESCRIPTION
**Pull request by jan-gogo**
Fixes #983 

The summary table gives the correct output getting the device name as `TEST AXIS VALUE` into account.
Use mixed devices configuration for testing.

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

File: test_runner/src/main/kotlin/ftl/json/MatrixMap.kt [hunk 1]
```diff
@@ -39,9 +39,9 @@ class MatrixMap(
      */
     fun validateMatrices(shouldIgnore: Boolean = false) {
         map.values.run {
-            firstOrNull { it.canceledByUser() }?.let { throw MatrixCanceledError(it.outcomeDetails.orEmpty()) }
-            firstOrNull { it.infrastructureFail() }?.let { throw InfrastructureError(it.outcomeDetails.orEmpty()) }
-            firstOrNull { it.incompatibleFail() }?.let { throw IncompatibleTestDimensionError(it.outcomeDetails.orEmpty()) }
+            firstOrNull { it.canceledByUser() }?.run { throw MatrixCanceledError(outcomeDetails) }
+            firstOrNull { it.infrastructureFail() }?.run { throw InfrastructureError(outcomeDetails) }
```
Comment (Sloox) [inline test_runner/src/main/kotlin/ftl/json/MatrixMap.kt:43]: nice change to make it more concise 👍

**Review by Sloox — APPROVED** [review]
Nice changes. Good that we managed to refactor Savedmatrix before this.

**Comment by jan-gogo** [comment]
> I think that text axis value should match gcloud
> `<device name>-<api level>-<locale>-<orientation>`
> because for now we just distinguish device name

```
┌─────────┬──────────────────────┬──────────────────────────┬───────────────────────────────┐
│ OUTCOME │      MATRIX ID       │     TEST AXIS VALUE      │         TEST DETAILS          │
├─────────┼──────────────────────┼──────────────────────────┼───────────────────────────────┤
│ flaky   │ matrix-32g83sldxamvi │ Nexus6P-26-en-portrait   │ 15 test cases passed, 5 flaky │
│ flaky   │ matrix-32g83sldxamvi │ greatqlte-26-en-portrait │ 19 test cases passed, 1 flaky │
└─────────┴──────────────────────┴──────────────────────────┴───────────────────────────────┘
```

File: test_runner/src/main/kotlin/ftl/reports/outcome/Util.kt [hunk 2]
```diff
@@ -1,19 +1,29 @@
...
-    ?.value.orUnknown()
+internal fun Step.deviceModel() = dimensionValue.deviceModel()
 
-internal fun Environment.deviceModel() = dimensionValue["Model"]
-    ?.value.orUnknown()
+internal fun Environment.deviceModel() = dimensionValue.deviceModel()
 
-operator fun List<StepDimensionValueEntry>?.get(key: String) =
-    this?.firstOrNull { it.key == key }
+private fun List<GenericJson>?.deviceModel() = this
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/reports/outcome/Util.kt:13]: should not be `textAxisValue` cause now it contains more than deviceModel?

**Review by piotradamczyk5 — APPROVED** [review]

**Pull request merged**